### PR TITLE
BENCH: simplify benchmark setup, remove benchmarks/run.py

### PR DESCRIPTION
**Pull request by rgommers**
This is a very belated follow-up to https://github.com/scipy/scipy/issues/4789#issuecomment-97829621 to make it more understandable what running the benchmarks actually does.

Autodetection of Python version and tool to use for environment creation should make benchmarks easier to run in a variety of build setups.
    
Also update dependencies:
- `six` and `Tempita` are no longer dependencies
- upgrade Cython to latest (0.29.18 isn't in conda defaults).

### CODE REVIEW

**Comment by mdhaber** [comment]
Do you suggest running with `runtests.py` or with `asv` directly?

**Comment by tylerjereddy** [comment]
I always use `asv` directly myself

File: benchmarks/asv.conf.json
Comment (mdhaber) [inline benchmarks/asv.conf.json:33]: Glad to see you [changed your mind](https://github.com/scipy/scipy/pull/10943#issuecomment-573756052)!

File: benchmarks/asv.conf.json
Comment (mdhaber) [inline benchmarks/asv.conf.json:59]: Looks helpful.

**Review by mdhaber — APPROVED** [review]
This also seems to fix the `TypeError: environment can only contain strings` error I [get on Windows](https://github.com/scipy/scipy/pull/12732#issuecomment-674595399). Combined with not having to edit the Python version in `asv.conf.json` in order to get benchmarks to run, this is really helpful.  One suggestion would be to update the [benchmark documentation](http://scipy.github.io/devdocs/dev/contributor/benchmarking.html#benchmarking-with-asv). rgommers/scipy#15

File: runtests.py
Comment (perimosocordiae) [inline runtests.py:250]: `run_asv` returns the result of `subprocess.call` or the literal `1`, but we ignore the return value here. Perhaps we should do `sys.exit(run_asv(cmd))` ?

File: benchmarks/asv.conf.json
Comment (rgommers) [reply benchmarks/asv.conf.json:59]: Agreed. I don't understand why asv devs would want to get rid of autodetection. Assuming users use a particular type of environment is very annoying for those users who are using a different env management method.

**Comment by rgommers** [comment]
> Do you suggest running with `runtests.py` or with `asv` directly?

I think we follow what we do with tests - `runtests.py` is documented as the default method, because it provides some consistency and tries to make things easier for new contributors. Both should work though. I personally don't use `runtests.py`.

File: runtests.py
Comment (rgommers) [reply runtests.py]: Good catch, fixed.

**Comment by rgommers** [comment]
The one CI failure is unrelated. Everyone seems happy here, so I'll aim to merge this soon unless there are more comments.

**Comment by mdhaber** [comment]
Please consider rgommers/scipy#15 before merging.

**Comment by rgommers** [comment]
Forgot to hit the green button on this, it has been good to go for a while. Merging.

**Pull request merged**
